TeslaCommand.go

```diff
@@ -1,179 +1,209 @@
+	flag.IntVar(&checkInterval, "checkInterval", 300, "Time in seconds between checks for geoFence")
+	flag.IntVar(&alertThreshold, "alertThreshold", 50, "Percentage charged threshold to send alert. If charge level is above threshold, alert won't be sent")
+}
+
+func main() {
+	// Setup Logging
+	logFileName := fmt.Sprintf("TeslaCommand-%v.log", time.Now().Unix())
+	logf, err := os.OpenFile(logFileName, os.O_WRONLY|os.O_CREATE, 0640)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	defer logf.Close()
+
+	log.SetOutput(logf)
+	logger := log.New(io.MultiWriter(logf, os.Stdout), "TeslaCommand: ", log.Lshortfile|log.LstdFlags)
+
+	logger.Printf("Num Args %d\n", len(os.Args))
+	flag.Parse()
+	if len(os.Args) != 20 {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+
+	client, err := tesla.NewClient(&tesla.Auth{ClientID: clientid, ClientSecret: clientsecret, Email: teslaLoginEmail, Password: teslaLoginPassword})
+	if err != nil {
+		logger.Println(err)
+		os.Exit(1)
+	}
+	//logger.Println("token: " + li.Token)
+
+	vehicles, err := client.Vehicles()
+	if err != nil {
+		logger.Println(err)
+		os.Exit(1)
+	}
+	vehicle := vehicles[vehicleIndex]
+	//LSE. Add vehicle wakeup call
+	_, err = vehicle.Wakeup()
+	if err != nil {
+		logger.Println(err)
+		os.Exit(1)
+	}
+	time.Sleep(60 * time.Second) //Give it a chance to wake up   LSE
+	// LSE
+	
+
+	vehicleState, err := vehicle.VehicleState()
+	if err != nil {
+		logger.Println(err)
+		os.Exit(1)
+	}
+
+	// Need to set this flag for every time vehicle exits and enters GeoFence (so we don't send repeated alerts)
+	// LSEingeofenceandstopped := false
+	waitmessage := fmt.Sprintf("Waiting to check vehicle %v location for %v seconds...\n", vehicleState.VehicleName, checkInterval)
+
+	// Loop every N seconds.
+	logger.Printf(waitmessage)
+	for _ = range time.Tick(time.Duration(checkInterval) * time.Second) {
+	
+	//LSE. Add vehicle wakeup call
+	_, err = vehicle.Wakeup()
+	if err != nil {
+		logger.Println(err)
+		os.Exit(1)
+	}
+	time.Sleep(60 * time.Second) //Give it a chance to wake up   LSE
+	// LSE
+	
+		logger.Printf("Checking vehicle %v location after waiting %v seconds.\n", vehicleState.VehicleName, checkInterval)
+
+		driveState, err := vehicle.DriveState()
+		if err != nil {
+			logger.Println(err)
+			logger.Printf(waitmessage)
+			continue
+		}
+
+		distance := HaversinFormula.Distance(geoFenceLatitude, geoFenceLongitude, driveState.Latitude, driveState.Longitude)
+
+		logger.Printf("Distance: %v\n\n", distance)
+
+		// If the distance is outside the radius, that means vehicle is outside the GeoFence.  Ok to get out
+		if distance > float64(radius) {
+			// LSEingeofenceandstopped = false
+			logger.Printf(waitmessage)
+			continue
+		}
+
+        // The following code was removed in order to make loop do the whole test each time, including sending email if disconnected LSE
+		// This is to prevent the below logic, if it's already executed, no need to keep doing it
+		// LSEif ingeofenceandstopped == true {

```

Comment:
Remove

Reply:
If within the geofence and disconnected, second+ time through will not send notification with this code in place. I want long check intervals (to prevent battery decline) with a message each time.